src/toil/wdl/wdltoil.py

```diff
@@ -2204,8 +2236,53 @@ def patched_run_invocation(*args: Any, **kwargs: Any) -> List[str]:
                     return command_line
 
                 # Apply the patch
-                task_container._run_invocation = patched_run_invocation #  type: ignore
+                task_container._run_invocation = patched_run_invocation  # type: ignore
+
+                singularity_original_prepare_mounts = task_container.prepare_mounts
+
+                def patch_prepare_mounts_singularity() -> List[Tuple[str, str, bool]]:
+                    """
+                    Mount the mount points specified from the disk requirements.
 
+                    The singularity and docker patch are separate as they have different function signatures
+                    """
+                    # todo: support AWS EBS/Kubernetes persistent volumes
+                    # this logic likely only works for local clusters as we don't deal with the size of each mount point
+                    mounts: List[Tuple[str, str, bool]] = singularity_original_prepare_mounts()
+                    # todo: support AWS EBS/Kubernetes persistent volumes
+                    # this logic likely only works for local clusters as we don't deal with the size of each mount point
+                    for mount_point, _ in self._mount_spec.items():
+                        abs_mount_point = os.path.abspath(mount_point)
+                        mounts.append((abs_mount_point, abs_mount_point, True))
```

Comment:
How sure are you that the spec actually means you're supposed to mount this particular host path through to the container?

It does say that the host path is required to exist. But it also says a lot about mounting "persistent volumes" *at* these mount points, and making them the required size. It doesn't seem to say anything about mounting those host paths specifically into the container.

What if you ask for a 100 GiB `/tmp` mount, and `/tmp` exists on the host, but `/tmp` on the host is only 10 GiB and Toil is actually doing all its work on a much larger `/scratch/tmp`? Shouldn't you get a 100 GiB `/tmp` mount in the container that actually refers to some location in `/scratch/tmp` on the host?

If the mount point feature was really meant to mount particular host paths across, wouldn't it take both a host-side path and a container-side path like the actual underlying mount functionality uses?